packages/forklift-console-plugin/src/modules/Providers/views/details/components/CredentialsSection/components/edit/EsxiCredentialsEdit.tsx

```diff
@@ -89,7 +89,7 @@ export const EsxiCredentialsEdit: React.FC<EditComponentProps> = ({ secret, onCh
       // don't trim fields that allow spaces
       const encodedValue = ['cacert'].includes(id)
         ? Base64.encode(value)
-        : Base64.encode(value.trim());
+        : Base64.encode(value?.trim());
```

Comment:
```suggestion
        : Base64.encode(value?.trim() || '');
```
if we expect `value` to be `undefined` we need to address it explicitly,  `Base64.encode(undefined)` has unexpected value, in the case of  `undefined` we need to decide to we want to set the encoded value to '', or `undefined` in the case of '' we already have code that translate '' to undefined when we save the data, so I will suggest going with `|| ''`